design/host-cgroups.md

```diff
@@ -0,0 +1,230 @@
+# Host cgroup management
+
+## Introduction
+
+Kata Containers runs over two layers of cgroups, the first layer is in the
+guest where only the workload is placed, the second layer is in the host and it
+will be explained on this document.
+
+```
+$ docker run -dt --runtime=kata-runtime nginx
+.-----------------------------------.
+|   .-------.                        |
+|   | nginx |                        |
+|.--'-------'---.  .------------.    |
+|| Guest Cgroup |  | Kata agent |    | <- Second cgroup layer (guest level)
+|--------------'--'-------------'    |
+|                                    |
+| .'-------------------------------. |
+| | Threads: vcpus + io trheads    | |
+| '--------------------------------. | .--------------.
+| Hypervisor                         | |shim + proxy  |
+.------------------------------------'---'--------------.
+|                  Kata processsses                     |
+.-------------------------------------------------------'
+|                   host Cgroups                        | <- First cgroup layer (host level)
+'-------------------------------------------------------'
+```
+
+In Kata Containers, workloads run in a virtual machine that is managed by a virtual
+machine monitor (VMM) running on the host. As a result, Kata Containers run over two layers of cgroups. The
+first layer is in the guest where the workload is placed, while the second layer is on the host where the
+VMM and associated threads are running.
+
+The OCI [runtime specification][linux-config] provides guidance on where the container cgroups should be placed:
+
+- [`CgroupsPath`][cgroupspath]
+> `cgroupsPath` (string, OPTIONAL) path to the cgroups. It can be used to either control the cgroups
+> hierarchy for containers or to run a new process in an existing container
+
+cgroups are hierarchical, and this can be seen with the following pod example:
+
+- Pod 1: `cgrouppath=/kubepods/pod1`
+  - Container 1:
+`cgrouppath=/kubepods/pod1/container1`
+  - Container 2:
+`cgrouppath=/kubepods/pod1/container2`
+
+- Pod 2: `cgrouppath=/kubepods/pod2`
+  - Container 1:
+`cgrouppath=/kubepods/pod2/container2`
+  - Container 2:
+`cgrouppath=/kubepods/pod2/container2`
+
+Depending on the upper-level orchestrator, the cgroup under which the pod is placed is
+managed by the orchestrator. In the case of Kubernetes, the pod-cgroup is created by Kubelet,
+while the container cgroups are to be handled by the runtime. Kubelet will size the pod-cgroup
+based on the container resource requirements.
+
+Kata Containers introduces a non-negligible overhead for running a sandbox (pod). Based on this, two scenarios are possible:
+ 1) The upper-layer orchestrator takes the overhead of running a sandbox into account when sizing the pod-cgroup, or
+ 2) Kata Containers do not fully constrain the VMM and associated processes, instead placing a subset of them outside of the pod-cgroup.
+
+Kata Containers provides two options for how cgroups are handled on the host.
```

Comment:
Obvious question that a newby might ask: "where are these settings? How do I enable them?" We should point to https://github.com/kata-containers/runtime#configuration.